### PR TITLE
Implement idling resources for tests.

### PR DESCRIPTION
**Pull request by m-sasha**
## Proposed Changes

Implement support for [idling resources](https://developer.android.com/jetpack/compose/testing#idling-resources) in `SkikoComposeUiTest`.
Note that the typical way idling resources are used requires `sleep` (and doesn't make sense in a single-threaded environment), so I added it as an expect/actual, with the JS implementation raising an exception.

We can consider supporting idling resources for JS, but it would require a different API.

## Testing

Test: Added a unit test.

### CODE REVIEW

File: compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/BasicTestTest.kt [hunk 1]
```diff
@@ -137,4 +140,45 @@ class BasicTestTest {
             }
         }
     }
+
+    @Test
+    fun testIdlingResource() {
```
Comment (eymar) [inline compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/BasicTestTest.kt:145]: Could this test live in "skikoTest" with `@IgnoreJsTarget` to run it for k/native too?
In skikoTests, instead of the rule there is a method "runSkikoComposeUiTest" in the scope of which `setContent` and `registerIdlingResource ` should be available.
If more tests from in this file can be in "skikoTests", then maybe it's better to do it in a separate PR.

File: compose/ui/ui-test-junit4/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skikoMain.kt [hunk 2]
```diff
@@ -243,6 +257,9 @@ class SkikoComposeUiTest(
         while (!isIdle()) {
             renderNextFrame()
             uncaughtExceptionHandler.throwUncaught()
+            if (!areAllResourcesIdle()) {
+                delay(IDLING_RESOURCES_CHECK_INTERVAL_MS)
```
Comment (eymar) [inline compose/ui/ui-test-junit4/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skikoMain.kt:261]: so JS tests should avoid using "waitForIdle" and use "awaitIdle" instead within TestScope (coroutine scope). Then idling resources should work there too, right?

**Review by eymar — APPROVED** [review]
LGTM! the comments are just questions/suggestions.

File: compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/BasicTestTest.kt (hunk 1 above)
Comment (m-sasha) [reply compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/BasicTestTest.kt:145]: Right now I'm working on replicating Android's test API (which uses `rule`), so I prefer to test with that. Later on, we can work on commonizing the test API, and we'll probably move/copy some tests into common (without `rule`).

File: compose/ui/ui-test-junit4/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skikoMain.kt (hunk 2 above)
Comment (m-sasha) [reply compose/ui/ui-test-junit4/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skikoMain.kt:261]: That's what I was thinking, but the problem is that all the automatic synchronization (following `setContent` and when you do `onNode(...)`) calls `waitForIdle`. So to use it with JS we'll need a new, async, API for those calls.
